ckan/templates/macros/form.html

```diff
@@ -439,6 +438,19 @@
     {% endif %}
     {% endif %}
 
-    {% if is_upload_enabled %}</div>{% endif %}
+  {% if query %}
+    {% call input_block('query', label=_('Query'), error='', classes=['control-full', 'editor']) %}
+      <textarea id="field-query" name="query" placeholder='SELECT * FROM "23af..."' rows="5" cols="20" class="form-control">{{ data.get('query', '') | empty_and_escape }}</textarea>
+      <span class="editor-info-block">{% trans %}Create a snapshot of data
+        from other DataStore tables using an SQL Query. You may re-run this
+        query at any time from the resource preview screen.<br/>
+        <b>Note:</b> This query will run
+        with your permissions but the data will be published and visible
+        along with other data in this dataset.{% endtrans %}
```

Comment:
Does this mean that considering

* Dataset A, public
* Dataset B, private
* User X, with admin permissions on Dataset B

If user X is creating a resource query in dataset A that includes data from a resource in dataset B, the data from that new resource query will be publicly accessible under dataset A?

If so there's a potential for users exposing private data without realizing. The message should reflect this better, and there should be a warning in the documentation (`datastore.rst`)

Reply:
Agreed, can you suggest some better text?

A query could be used to publish from a private table while removing all PII information, so this isn't always a drawback. Also there's nothing new being allowed here that couldn't already be done by the same user with `datastore_search_sql` and `datastore_upsert`.

Reply:
> Also there's nothing new being allowed here that couldn't already be done by the same user with datastore_search_sql and datastore_upsert

True, but this feature makes it much easier to do, and so more prone to accidents.

What about something like:

> The query will be run using your permissions but the data will be published using this dataset visibility settings, regardless of the permissions on other resources used. Keep this in mind when querying private datasets.

Reply:
Sounds much better!